### ISSUE TITLE
Feature collapse child window fit content

### ISSUE THREAD

**Issue by xpenatan** [ocornut/imgui#2779]
Hi, I wanted to have a implementation similar to CollapsingHeader but with my own custom frame (Example: have multiple buttons, custom style,  border, etc). And also having multiple collapsing child layouts.  

CollapsingHeader  use a tree and I first tried to replicate it for my custom needs without success.  I wanted it to have its own layout so nothing outside interfere with its content.  

Im new to custom widgets so I dont know if using only groups or mainipulating window is the right way so I starting analysing the child windows methods.  But the problem is that there is no fit implementation. I wanted the child window to only grow its height when the child contents gets bigger and not using -1 or 0 for height. Its similar to #1496   

I digged in and manipulated child window to have this feature by calculaing and changing its height with a ending method call. With it I could create a custom frame which could enable/disable the window draw calls. Its my first try so its possible that there is a better way to do it.

All examples have a debug window which shows the padding area (red) and the content area (green).

Example 1: Using custom child window.

![example01](https://user-images.githubusercontent.com/6472084/64878212-651e5280-d629-11e9-8575-65f1f39f6543.gif)

Example 2: Using a custom collapse header with border and calling the custon child window.

![example02](https://user-images.githubusercontent.com/6472084/64878488-073e3a80-d62a-11e9-87b0-be2f5e90eb9b.gif)

Example 3: Using multiple child window.  The second window respect the parent window width and grows its height when it needs space.

![example03](https://user-images.githubusercontent.com/6472084/64878596-3fde1400-d62a-11e9-89a7-6f68afecb512.gif)

Code sample:
``` c++
 static bool isOpen= false;
 if(ImGui::BeginCollapsingHeaderEx("idChild", "Title", &isOpen, 0, 0, 0))
            renderContent();
 ImGui::EndCollapsingHeaderEx(isOpen);
```

Widget class file:
https://gist.github.com/xpenatan/d3a5dd97d3b767258237939ef6377907

To test include "imgui_custom_widgets.h" and call ImGui::ShowCustomWindowDemo()  to see the examples.

Is it possible to have this layout feature for child window where its height only gets bigger when its contents need space ?

**Comment by ocornut** [comment]
I don't understand why you are using child windows here. The purpose of child windows is to allow for an individual scrolling regions. If you auto-fit them you are not using that feature and therefore you probably don't need a child window at all. What's the reasoning behind this?

**Comment by xpenatan** [comment]
Hi, I reworked the UI.  Removed the child window and created a "ImLayout" object to support what I need. Also implemented a simple alignment feature to be easy to position widgets. Ex left, center, right, top, bottom, etc.

If a similar feature is implemented to the core it will be great but for now I'll use this custom widget.

![example012](https://user-images.githubusercontent.com/6472084/65841548-e4827600-e2f9-11e9-81dc-127ecdf79d66.gif)

If anyone wants to try it out the code is here: 
https://github.com/xpenatan/jDear-imgui/tree/master/extensions/imgui-layout-widget
